plugins/pulp_ostree/plugins/importers/steps.py

```diff
@@ -180,3 +183,205 @@ def link(self, unit):
                 pass  # identical
             else:
                 raise
+
+
+class Clean(PluginStep):
+    """
+    Clean up after import.
+    """
+
+    def __init__(self):
+        super(Clean, self).__init__(step_type=constants.IMPORT_STEP_CLEAN)
+        self.description = _('Clean')
+
+    def process_main(self):
+        """
+        Clean up after import:
+         - Delete the remote used for the pull.
+        """
+        path = self.parent.storage_path
+        remote_id = self.parent.repo_id
+        try:
+            repository = lib.Repository(path)
+            remote = lib.Remote(remote_id, repository)
+            remote.delete()
+        except lib.LibError, le:
+            pe = PulpCodedException(errors.OST0003, id=remote_id, reason=str(le))
+            raise pe
+
+
+class Remote(object):
+    """
+    Represents an OSTree remote.
+    Used to build and configure an OSTree remote.
+    The complexity of configuring the remote based on the importer
+    configuration is isolated here.
+
+    :ivar step: The create step.
+    :type step: Create
+    :ivar repository: An OSTree repository.
+    :type repository: lib.Repository
+    """
+
+    def __init__(self, step, repository):
+        """
+        :param step: The create step.
+        :type step: Create
+        :param repository: An OSTree repository.
+        :type repository: lib.Repository
+        """
+        self.step = step
+        self.repository = repository
+
+    @property
+    def url(self):
+        """
+        The remote URL.
+
+        :return: The remote URL
+        :rtype: str
+        """
+        return self.step.parent.feed_url
+
+    @property
+    def remote_id(self):
+        """
+        The remote ID.
+
+        :return: The remote ID.
+        :rtype: str
+        """
+        return self.step.parent.repo_id
+
+    @property
+    def working_dir(self):
+        """
+        The working directory.
+
+        :return: The absolute path to the working directory.
+        :rtype: str
+        """
+        return self.step.get_working_dir()
+
+    @property
+    def config(self):
+        """
+        The importer configuration.
+
+        :return: The importer configuration.
+        :rtype: pulp.server.plugins.config.PluginCallConfiguration
+        """
+        return self.step.get_config()
+
+    @property
+    def ssl_key_path(self):
+        """
+        The SSL private key path.
+
+        :return: The absolute path to the private key.
+        :rtype: str
+        """
+        path = None
+        key = self.config.get(importer_constants.KEY_SSL_CLIENT_KEY)
+        if key:
+            path = os.path.join(self.working_dir, 'key.pem')
+            with open(path, 'w+') as fp:
```

Comment:
Will this result in a world readable private key? I believe the default permissions for created files with `open` are 644 (but I couldn't find documentation stating this, just my own tests).